src/app/features/search-bar/search-bar.component.ts

```diff
@@ -0,0 +1,276 @@
+      tap(() => this.isLoading$.next(false)),
+    );
+
+    this._attachKeyDownHandlerTimeout = window.setTimeout(() => {
+      this.inputEl.nativeElement.addEventListener('keydown', (ev: KeyboardEvent) => {
+        if (ev.key === 'Escape') {
+          this.blurred.emit();
+        } else if (ev.key === '1' && ev.ctrlKey) {
+          this.switchTaskSource();
+          ev.preventDefault();
+        } else if (
+          ev.key === 'Enter' &&
+          (!this.taskSuggestionsCtrl.value ||
+            typeof this.taskSuggestionsCtrl.value === 'string')
+        ) {
+          this._shakeSearchForm();
+          ev.preventDefault();
+        }
+      });
+    });
+  }
+
+  private _getLocation(item: SearchItem): string {
+    const tasksOrWorklog = !this.isArchivedTasks ? 'tasks' : 'worklog';
+    if (item.projectId) {
+      return `/project/${item.projectId}/${tasksOrWorklog}`;
+    } else if (item.tagIds.includes(TODAY_TAG.id)) {
+      return `/tag/TODAY/${tasksOrWorklog}`;
+    } else if (item.tagIds[0]) {
+      return `/tag/${item.tagIds[0]}/${tasksOrWorklog}`;
+    } else {
+      devError("Couldn't find task location");
+      return '';
+    }
+  }
+
+  private _getArchivedDate(item: SearchItem, tasks: Task[]): string {
+    let dateStr = Object.keys(item.timeSpentOnDay)[0];
+    if (dateStr) return dateStr;
+
+    if (item.parentId) {
+      const parentTask = tasks.find((task) => task.id === item.parentId) as Task;
+      dateStr = Object.keys(parentTask.timeSpentOnDay)[0];
+      return dateStr ?? getWorklogStr(parentTask.created);
+    }
+
+    return getWorklogStr(item.created);
+  }
+
+  private _isInBacklog(item: SearchItem, projects: Project[]): boolean {
+    if (!item.projectId) return false;
+    const project = projects.find((p) => p.id === item.projectId);
+    return project ? project.backlogTaskIds.includes(item.id) : false;
+  }
+
+  private _filter(searchableItems: SearchItem[], searchTerm: string): SearchItem[] {
+    let result = searchableItems.filter(
+      (task) =>
+        task.title.includes(searchTerm.toLowerCase()) ||
+        task.taskNotes.includes(searchTerm.toLowerCase()),
+    );
+
+    if (result.length > 200) {
+      this.tooManyResults = true;
+      result = result.slice(0, 200);
+    } else {
+      this.tooManyResults = false;
+    }
+
+    return result;
+  }
+
+  private _shakeSearchForm(): void {
+    this.searchForm.nativeElement.classList.toggle('shake-form');
+    this.searchForm.nativeElement.onanimationend = () => {
+      this.searchForm.nativeElement.classList.toggle('shake-form');
+    };
+  }
+
+  switchTaskSource(event?: MouseEvent) {
+    // trigger on mousedown to keep inputEl in focus
+    event?.preventDefault();
+    this.isLoading$.next(true);
+    this.isArchivedTasks = !this.isArchivedTasks;
+    this.isArchivedTasks$.next(this.isArchivedTasks);
+    if (IS_MOBILE) {
+      this._openPanelTimeout = window.setTimeout(() => this.autocomplete.openPanel());
+    }
```

Comment:
For some reason the autocomplete list closed after switching task source on mobile browser. This way it will stay open.